firmware/version_data.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+
+import subprocess
+import os
+import tempfile
+import filecmp
+import shutil
+
+commit = subprocess.check_output(['git', 'log', '--format="%H"', '-n', '1'])
```

Comment:
With Python 3 you want to explicitly decode the output from iostreams (like from subprocess).

```python
stdout = subprocess.check_output(...).decode('utf-8')
```